consul/resource_consul_config_entry_service_defaults.go

```diff
@@ -0,0 +1,683 @@
+				if i == 0 {
+					return nil
+				}
+				ui := uint32(i)
+				return &ui
+			}
+			passiveHealthCheck := &consulapi.PassiveHealthCheck{
+				MaxFailures:             uint32(passiveHealthCheckMap["max_failures"].(int)),
+				EnforcingConsecutive5xx: uint32Ptr(passiveHealthCheckMap["enforcing_consecutive_5xx"].(int)),
+				MaxEjectionPercent:      uint32Ptr(passiveHealthCheckMap["max_ejection_percent"].(int)),
+			}
+			duration, err := time.ParseDuration(passiveHealthCheckMap["interval"].(string))
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse interval: %w", err)
+			}
+			passiveHealthCheck.Interval = duration
+			baseEjectionTime, err := time.ParseDuration(passiveHealthCheckMap["base_ejection_time"].(string))
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse base_ejection_time: %w", err)
+			}
+			passiveHealthCheck.BaseEjectionTime = &baseEjectionTime
+			return passiveHealthCheck, nil
+		}
+		return nil, nil
+	}
+
+	getMeshGateway := func(meshGateway interface{}) *consulapi.MeshGatewayConfig {
+		meshGatewayList := meshGateway.(*schema.Set).List()
+		if len(meshGatewayList) > 0 {
+			meshGatewayData := meshGatewayList[0].(map[string]interface{})
+			return &consulapi.MeshGatewayConfig{
+				Mode: consulapi.MeshGatewayMode(meshGatewayData["mode"].(string)),
+			}
+		}
+		return nil
+	}
+
+	getUpstreamConfig := func(upstreamConfigMap map[string]interface{}) (*consulapi.UpstreamConfig, error) {
+		upstreamConfig := &consulapi.UpstreamConfig{
+			Name:              upstreamConfigMap["name"].(string),
+			Partition:         upstreamConfigMap["partition"].(string),
+			Namespace:         upstreamConfigMap["namespace"].(string),
+			Peer:              upstreamConfigMap["peer"].(string),
+			EnvoyListenerJSON: upstreamConfigMap["envoy_listener_json"].(string),
+			EnvoyClusterJSON:  upstreamConfigMap["envoy_cluster_json"].(string),
+			Protocol:          upstreamConfigMap["protocol"].(string),
+			ConnectTimeoutMs:  upstreamConfigMap["connect_timeout_ms"].(int),
+			Limits:            getLimits(upstreamConfigMap["limits"].(*schema.Set).List()[0].(map[string]interface{})),
+		}
+		passiveHealthCheck, err := getPassiveHealthCheck(upstreamConfigMap["passive_health_check"])
+		if err != nil {
+			return nil, err
+		}
+		upstreamConfig.PassiveHealthCheck = passiveHealthCheck
+		upstreamConfig.MeshGateway = *getMeshGateway(upstreamConfigMap["mesh_gateway"])
+		upstreamConfig.BalanceOutboundConnections = upstreamConfigMap["balance_outbound_connections"].(string)
+		return upstreamConfig, nil
+	}
+
+	getTransparentProxy := func(transparentProxy map[string]interface{}) *consulapi.TransparentProxyConfig {
+		return &consulapi.TransparentProxyConfig{
+			OutboundListenerPort: transparentProxy["outbound_listener_port"].(int),
+			DialedDirectly:       transparentProxy["dialed_directly"].(bool),
+		}
+	}
+
+	getEnvoyExtension := func(envoyExtensionMap map[string]interface{}) *consulapi.EnvoyExtension {
```

Comment:
```suggestion
	getEnvoyExtension := func(envoyExtensionMap map[string]interface{}) consulapi.EnvoyExtension {
```

You can return a `consulapi.EnvoyExtension` here because it is what you need later.